network/tcp.go

```diff
@@ -0,0 +1,411 @@
+package network
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/dedis/cothority/log"
+)
+
+// NewTCPRouter returns a new Router using TCPHost as the underlying Host.
+func NewTCPRouter(sid *ServerIdentity) (*Router, error) {
+	h, err := NewTCPHost(sid.Address)
+	if err != nil {
+		return nil, err
+	}
+	r := NewRouter(sid, h)
+	return r, nil
+}
+
+// TCPConn implements the Conn interface using plain, unencrypted TCP.
+type TCPConn struct {
+	// The name of the endpoint we are connected to.
+	endpoint Address
+
+	// The connection used
+	conn net.Conn
```

Comment:
This also is only nitpicking and since this isn't exported it's OK to leave it as it is. I just like to be more explicit. Keeping it a `net.Conn` has the advantage of being able to mock this with another implementation (for testing purposes).